pandas/core/generic.py

```diff
@@ -9961,14 +9990,75 @@ def _check_percentile(self, q):
               2016      2015      2014
         GOOG   NaN -0.151997 -0.086016
         APPL   NaN  0.337604  0.012002
+
+        In a DataFrame with NAs, when computing the percentage change with
+        default `skipna=True`, NAs are first droppped on each column/row, and
+        the results are eventually reindexed as originally.
+
+        >>> df = pd.DataFrame({
+        ...     'a': [90, 91, None, 85, None, 95],
+        ...     'b': [91, None, 85, None, 95, None],
+        ...     'c': [None, 85, None, 95, None, None]})
+        >>> df
+              a     b     c
+        0  90.0  91.0   NaN
+        1  91.0   NaN  85.0
+        2   NaN  85.0   NaN
+        3  85.0   NaN  95.0
+        4   NaN  95.0   NaN
+        5  95.0   NaN   NaN
+
+        >>> df.pct_change()
+                  a         b         c
+        0       NaN       NaN       NaN
+        1  0.011111       NaN       NaN
+        2       NaN -0.065934       NaN
+        3 -0.065934       NaN  0.117647
+        4       NaN  0.117647       NaN
+        5  0.117647       NaN       NaN
+
+        >>> df.pct_change(axis=1)
+            a         b         c
+        0 NaN  0.011111       NaN
+        1 NaN       NaN -0.065934
+        2 NaN       NaN       NaN
+        3 NaN       NaN  0.117647
+        4 NaN       NaN       NaN
+        5 NaN       NaN       NaN
+
+        Otherwise, if a fill method is passed, NAs are filled before the
+        computation.
+
+        >>> df.pct_change(fill_method='ffill')
+                  a         b         c
+        0       NaN       NaN       NaN
+        1  0.011111  0.000000       NaN
+        2  0.000000 -0.065934  0.000000
+        3 -0.065934  0.000000  0.117647
+        4  0.000000  0.117647  0.000000
+        5  0.117647  0.000000  0.000000
         """
 
     @Appender(_shared_docs['pct_change'] % _shared_doc_kwargs)
-    def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
-                   **kwargs):
-        # TODO: Not sure if above is correct - need someone to confirm.
+    def pct_change(self, periods=1, fill_method=None, limit=None, freq=None,
+                   skipna=None, **kwargs):
+        if fill_method is not None and skipna:
+            raise ValueError("cannot pass both fill_method and skipna")
+        elif limit is not None and skipna:
+            raise ValueError("cannot pass both limit and skipna")
```

Comment:
it would more clear if we an actually default ``skipna=True``, is that probelmatic?

Reply:
Yes, see what I previously commented to you: 
- https://github.com/pandas-dev/pandas/pull/25291#discussion_r257833942

> @jreback I have a question about setting `skipna=True` as default. Then, in order to use the parameter `fill_method`, the user should pass `skipna=False` explicitly, so that no ValueError is raised. For example:
> 
> ```python
> df.pct_change(fill_method='bfill', limit=2, skipna=False)
> ```
> 
> I do not see this very intuitive. So my question is: should we keep the parameters `fill_method` and `limit` in this method? Or should we better advise to use `fillna` before calling `pct_change`?
> For example:
> 
> ```python
> df.fillna(method='bfill', limit=2).pct_change()
> ```


- https://github.com/pandas-dev/pandas/pull/25291#discussion_r258711552

> Finally, I have set default `skipna=True` only if `fill_method` and `limit` are None.